### DISCUSSION TITLE
change standard formatting for JS

### DISCUSSION THREAD

**Discussion by tobsn** [zed-industries/zed#8217] [Feature Requests]
Is there a way to edit the lsp settings for js formatting? It's a bit too aggressive to me...
All I found is how to add a different formatter like prettier etc.

thanks!

**Comment by SomeoneToIgnore** [comment]
See general overview of Zed's formatter capabilities and the settings: https://github.com/zed-industries/zed/discussions/7951#discussioncomment-8501391

Quick recap:

First, you have to understand what is a `formatter` setting in your settings set to.
If there's none, then the default value would be `auto`, which tries to format the buffer with `prettier` (if applicable to your current language which unfortunately is not mentioned in the question) or, otherwise, tries to format the buffer with the language server formatting.

Prettier is configured with `"prettier"` section with https://prettier.io/docs/en/configuration , each language server is configured differently and it's better to consult its docs.

**Comment by tobsn** [comment]
Oh.. that's too easy to understand at first ;)
So (or js) if you want to edit it you basically have to overwrite the default used "no settings" prettier settings via prettierrc - thats how?

**Reply by SomeoneToIgnore** [reply]
Sorry, the last sentence is a bit tricky to understand, but I would assume you have zero formatter-related settings in your settings.json and some language that is applicable to prettier formatting.

In that case, you have to add something like
`"prettier": {
  "trailingComma": "es5",
  "tabWidth": 4,
  "semi": false,
  "singleQuote": true
}` into your settings, but use the config, relevant to your needs, taken from https://prettier.io/docs/en/configuration or plugins' docs.

You can also call `debug: open language server logs` command and select prettier instance instance there to see if it takes effect or not:

<img width="1425" alt="image" src="https://github.com/zed-industries/zed/assets/2690773/bccf427c-25cb-490e-8607-c97cf2fb32c1">

**Reply by tobsn** [reply]
oh I see, so technically I can also always read out the prettier default settings via the debug, just to see how its set up "right now"?

**Reply by SomeoneToIgnore** [reply]
Prettier settings are empty by default, except for two settings: tabWidth and printWidth, that Zed sets based on its config — nothing else will be visible in that debug section.

https://prettier.io/docs/en/options.html may be helpful to understand what Prettier will use as defaults for empty settings.
